File: ruby/binary/binary.rb
module BookKeeping
  VERSION = 3
end

class Binary
  def self.to_decimal num
    raise ArgumentError if num.chars.any? { |i| !("0..1").include? i }
    count = num.length - 1
    sum = 0
    
    num.chars.each do |i| 
      sum += i.to_i * 2 ** count
      count -= 1
    end
    sum
  end
end

File: javascript/binary/binary.js
function Binary(num) {
  this.num = num;
}

Binary.prototype.toDecimal = function toDecimal() {
  if ((this.num.match(/[^01]/))) return 0;

  let sum = 0;
  let count = this.num.length - 1;

  for (let i = 0; i < this.num.length; i += 1) {
    sum += this.num[i] * (2 ** count);
    count -= 1;
  }
  return sum;
};

module.exports = Binary;
